packages/components/src/List/Toolbar/ColumnChooserButton/ColumnChooser/ColumnChooserRow/RowCheckbox/RowCheckbox.component.js

```diff
@@ -32,7 +32,7 @@ const RowCheckbox = ({
 				checked={checked}
 				data-feature={dataFeature}
 				aria-describedby={describedby}
-				id={`${id}-checkbox-${label}`}
+				id={`${id}-checkbox-${label.replace(/\s+/, '-')}`}
```

Comment:
This replaces only the first occurence, you can see it in the snapshot that was modified in this PR.
```suggestion
				id={`${id}-checkbox-${label.replace(/\s+/g, '-')}`}
```